### ISSUE TITLE
Reproduce results

### ISSUE THREAD

**Issue by phibenz** [locuslab/fast_adversarial#8]
Hello, 
thanks for the great work and open-sourcing the repository.
I reran the CIFAR10 experiments with the unmodified code (without arguments) provided and I got the following results:
`python train_fgsm.py:`
|       | Test Loss | Test Acc | PGD Loss | PGD Acc |
|-------|-----------|----------|----------|---------|
| My    | 0.6739    | 0.7930   | 1.0310   | 0.4531  |
| Paper | -         | 0.8381   | -        | 0.4606  |

`python train_free.py:`
|       | Test Loss | Test Acc | PGD Loss | PGD Acc |
|-------|-----------|----------|----------|---------|
| My    | 0.7544    | 0.7695   | 1.0670   | 0.4598  |
| Paper | -         | 0.7838   | -        | 0.4618  |

`python train_pgd.py:`
|       | Test Loss | Test Acc | PGD Loss | PGD Acc |
|-------|-----------|----------|----------|---------|
| My    | 0.7657    | 0.7664   | 1.0657   | 0.4725  |
| Paper | -         | 0.8246   | -        | 0.5069  |

Any hint how to close the performance gap between the reported results and the ones obtained with code (especially for `train_fgsm.py`)? 

I also have an additional question about Table 3 in the paper. Why is the time for the seconds/epoch of PGD-7 (1456.22) so much greater than DAWNBench + PGD-7 (104.94). From what I read online the speed improvements of mixed precision are usually in the range of 20% to 30%. Here it seems to increase the speed much more drastically. 

Thanks for your help

**Comment by leslierice1** [comment]
Hi, sorry for the delay in response here! The default arguments in the code here should correspond to the hyperparameter settings for just achieving the baseline 45% robust accuracy (i.e. Figure 2 in the paper). The only change you should need to make to close the gap is the number of epochs you train for - 30 epochs for FGSM, 40 epochs for PGD, 12 epochs for Free training (see Table 7 in the appendix for these numbers, and divide the Free training epochs in the paper by 8, the number of minibatch replays). 

As for your question about Table 3, another speedup from the DAWNBench competition is a smaller architecture size; for our experiments we used a PreAct ResNet18 architecture whereas the PGD-7 in Madry et al. 2017 used a Wide ResNet 18-10, which is much larger.
